src/network/mcpe/NetworkSession.php

```diff
@@ -419,10 +418,31 @@ public function handleEncoded(string $payload) : void{
 	 * @throws PacketHandlingException
 	 */
 	public function handleDataPacket(Packet $packet, string $buffer) : void{
+
+		$playerName = $this->getPlayer()?->getName();
+		if (!isset($this->packetsReceived[$playerName])) {
+			$this->packetsReceived[$playerName] = [
+				'count' => 0,
+				'time' => time(),
+			];
+		}
+
+		$playerPackets = &$this->packetsReceived[$playerName];
+		$playerPackets['count']++;
```

Comment:
I think passing by reference causes confusion
(It can be split into two properties)